src/coreclr/tools/aot/ILCompiler.MetadataTransform/Internal/Metadata/NativeFormat/Writer/NativeMetadataWriter.cs

```diff
@@ -4,12 +4,13 @@
 using System;
 using System.IO;
 using System.Collections.Generic;
+using System.Linq;
 using System.Reflection;
 using Debug = System.Diagnostics.Debug;
 using ConditionalAttribute = System.Diagnostics.ConditionalAttribute;
-using Internal.LowLevelLinq;
```

Comment:
What's the impact of switching away from Internal.LowLevelLinq on ilc compiler binary size? Could you please share `ilc.exe` binary size before/after this PR?

Reply:
12685KB -> 12735KB, it seems that size-optimized LINQ is still larger than naive implementation?

Reply:
It's now 12681KB in latest commit.

Reply:
Yes, the size-optimized LINQ is still a ton of code. It just avoids non-linear code size growth.

What would it take to delete LowLevelLinq completely? I think we either want to keep it; or delete it completely and take some size hit. Deleting it partially is kind of the worst of both worlds.